src/org/usfirst/frc2130/visionTargetting/Robot.java

```diff
@@ -0,0 +1,263 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+
+import org.opencv.core.*;
+import org.opencv.core.Core.*;
+import org.opencv.features2d.FeatureDetector;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.*;
+import org.opencv.objdetect.*;
+
+
+/**
+ * The VM is configured to automatically run this class, and to call the
+ * functions corresponding to each mode, as described in the IterativeRobot
+ * documentation. If you change the name of this class or the package after
+ * creating this project, you must also update the manifest file in the resource
+ * directory.
+ */
+
+
+
+public class Robot extends IterativeRobot {
+	
+
+
+	NetworkTable table;
+
+    Command autonomousCommand;
+    
+	public VisionThread visionThread;
+	
+	public static  Object imgLock;
+	
+	public static double centerX;
+	
+	public static double run;
+	
+
+	private static final int IMG_WIDTH = 320;
+	private static final int IMG_HEIGHT = 240;
+
+	private boolean run1;
+    
+    public static OI oi;
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+    public static robotDrive robotDrive;
+
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+
+    /**
+     * This function is run when the robot is first started up and should be
+     * used for any initialization code.
+     */
+    @Override
+    public void robotInit() {
+    RobotMap.init();  
+    	
+    // Creates UsbCamera and MjpegServer [1] and connects them
+
+    CameraServer.getInstance().startAutomaticCapture();
+
+    // Creates the CvSink and connects it to the UsbCamera
+    CvSink cvSink = CameraServer.getInstance().getVideo();
+
+    // Creates the CvSource and MjpegServer [2] and connects them
+    CvSource outputStream = CameraServer.getInstance().putVideo("Blur", 640, 480); 
+       
+         
+        // BEGIN AUTOGENERATED CODE, SOURCE = ROBOTBUILDER ID = CONSTRUCTORS
+        robotDrive = new robotDrive();
+
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID = CONSTRUCTORS
+        // OI must be constructed after subsystems. If the OI creates Commands
+        //(which it very likely will), subsystems are not guaranteed to be
+        // constructed yet. Thus, their requires() statements may grab null
+        // pointers. Bad news. Don't move it.
+        oi = new OI();
+
+        // instantiate the command used for the autonomous period
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=AUTONOMOUS
+
+        autonomousCommand = new AutonomousCommand();
+
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=AUTONOMOUS
+        
+        CameraServer.getInstance().startAutomaticCapture();
+        UsbCamera camera = new UsbCamera("USB Camera", 0);
+        camera.setResolution(IMG_WIDTH, IMG_HEIGHT);
+        
+        CameraServer.getInstance().getVideo();
+       
+        new Thread(() -> {        
```

Comment:
Now we're starting to get into the meat of the program.  This line is very important because it is instructing the computer to create a new parallel thread and points to an anonymous function with no inputs.  Everything from here down to the closing brace will be kicked off in parallel.

While creating this thread makes sense for the original image example, I think we'll actually want to remove this in favor of the vision thread created a little bit farther below.